File: src/main/java/xyz/oribuin/skyblock/enums/FilterType.java
package xyz.oribuin.skyblock.enums;

import xyz.oribuin.skyblock.island.Island;

import java.util.List;

public enum FilterType {
    NONE,

    // Categories
    GENERAL,
    FARMS,
    PARKOUR,
    SHOPS,
    DESIGN;

    /**
     * Filter the list of islands by the given type.
     *
     * @param islands The list of islands to filter.
     * @return The filtered list of islands.
     */
    public List<Island> filter(List<Island> islands) {
        if (this == FilterType.NONE)
            return islands;

        islands.removeIf(island -> !island.getWarp().getCategory().getTypes().contains(this.name()));
        return islands;
    }

}

File: Plugin/src/main/kotlin/xyz/oribuin/skyblock/enums/FilterType.kt
package xyz.oribuin.skyblock.enums

import xyz.oribuin.skyblock.island.Island

enum class FilterType {
    NONE,

    // Categories
    GENERAL,
    FARMS,
    PARKOUR,
    SHOPS,
    DESIGN;

    /**
     * Filter the list of islands by the given type.
     *
     * @param islands The list of islands to filter.
     * @return The filtered list of islands.
     */
    fun filter(islands: MutableList<Island>): MutableList<Island> {
        if (this == NONE)
            return islands

        islands.removeIf { !it.warp.category.types.contains(this.name) }
        return islands
    }

}